src/wrapper.rs

```diff
@@ -374,3 +389,120 @@ impl Dxc {
         );
     }
 }
+
+#[derive(Debug)]
+pub struct DxcValidator {
+    inner: ComPtr<IDxcValidator>,
+    dxc: Dxc,
+    library: DxcLibrary,
+}
+
+impl Drop for DxcValidator {
+    fn drop(&mut self) {
+        // Release library explicitly (it depends on dxc)
+        self.library = DxcLibrary::new(ComPtr::new());
+    }
+}
+
+pub type DxcValidatorVersion = (u32, u32);
+
+// https://www.wihlidal.com/blog/pipeline/2018-09-16-dxil-signing-post-compile/
+impl DxcValidator {
+    fn new(inner: ComPtr<IDxcValidator>, dxc: Dxc, library: DxcLibrary) -> Self {
+        Self {
+            inner,
+            dxc,
+            library,
+        }
+    }
+
+    pub fn version(&self) -> Result<DxcValidatorVersion, HRESULT> {
+        // TODO:
+        /*iid!(pub IID_IDxcVersionInfo = 0xb04f5b50, 0x2059, 0x4f12, 0xa8, 0xff, 0xa1, 0xe0, 0xcd, 0xe1, 0xcc, 0x7e);
+        com_interface! {
+            interface IDxcVersionInfo: IUnknown{
+                iid: IID_IDxcVersionInfo,
+                vtable: IDxcVersionInfoVtbl,
+
+                fn get_version(major: *mut u32, minor: *mut u32) -> HRESULT;
+                fn get_flags(flags: *mut u32) -> HRESULT;
+            }
+        }*/
+        Ok((1, 3))
+    }
+
+    pub fn validate_slice(&mut self, data: &[u8]) -> Result<(Vec<u8>, String), HRESULT> {
```

Comment:
The return type here should probably closer match the one of DxcCompiler::compile, so returning a `Result<Vec<u8>, (DxcOperationResult, HRESULT)>` since right now you're returning `Ok` in error cases. And I think it's nicer to return the `DxcOperationResult` instead of a `String`.